src/main/java/github/scarsz/discordsrv/hooks/permissions/LuckPermsHook.java

```diff
@@ -33,21 +40,35 @@
 import net.luckperms.api.node.NodeType;
 import org.bukkit.Bukkit;
 import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.server.PluginDisableEvent;
 import org.bukkit.plugin.Plugin;
+import org.checkerframework.checker.nullness.qual.NonNull;
 
 import java.util.*;
 
-public class LuckPermsHook implements PluginHook {
+public class LuckPermsHook implements PluginHook, ContextCalculator<Player> {
+    private static final String CONTEXT_LINKED = "discordsrv:linked";
+    private static final String CONTEXT_BOOSTING = "discordsrv:boosting";
+    private static final String CONTEXT_ROLE = "discordsrv:role";
 
+    private final LuckPerms luckPerms;
     private final Set<EventSubscription<?>> subscriptions = new HashSet<>();
 
     public LuckPermsHook() {
-        subscriptions.add(LuckPermsProvider.get().getEventBus().subscribe(UserTrackEvent.class, event -> handle(event.getUser().getUniqueId())));
-        subscriptions.add(LuckPermsProvider.get().getEventBus().subscribe(NodeAddEvent.class, event -> handle(event, event.getNode(), true)));
-        subscriptions.add(LuckPermsProvider.get().getEventBus().subscribe(NodeRemoveEvent.class, event -> handle(event, event.getNode(), false)));
+        luckPerms = Bukkit.getServicesManager().load(LuckPerms.class);
+
+        // update events
+        subscriptions.add(luckPerms.getEventBus().subscribe(UserTrackEvent.class, event -> handle(event.getUser().getUniqueId())));
+        subscriptions.add(luckPerms.getEventBus().subscribe(NodeAddEvent.class, event -> handle(event, event.getNode(), true)));
+        subscriptions.add(luckPerms.getEventBus().subscribe(NodeRemoveEvent.class, event -> handle(event, event.getNode(), false)));
```

Comment:
Since contexts have a separate on/off switch (`DisabledPluginHooks`), it would make sense if group updates did too (this is used for 'instant' group synchronization) I'd suggest `LuckPerms-GroupUpdates` or something like that as a entry for `DisabledPluginHooks`

Reply:
I went ahead and added that: https://github.com/lucko/DiscordSRV/commit/a5bbabf0a16cf8d8cad74a5ed7d5571a8e8dcf6f